Utils/Dataflow/009_oracleConnector/Oracle2JSON.py

```diff
@@ -38,60 +36,140 @@ def main():
+        queries = {}
+        for (qname, f) in queries_cfg:
+            queries[qname] = {'file': config_path(f, result)}
+        result['queries'] = queries
+    except (IOError, ConfigParser.Error), e:
+        sys.stderr.write('Failed to read config file (%s): %s\n'
+                         % (config_file, e))
+        return None
+
+    # Optional parameters
+    if result['step_seconds'] > 0:
+        default_initial = datetime.utcfromtimestamp(0)
+        default_final = None
+    else:
+        default_initial = datetime.now()
+        default_final = datetime.utcfromtimestamp(0)
+
+    result['final_date'] = str2date(config_get(config, "timestamps", "final",
+                                               default_final))
+    result['initial_date'] = str2date(config_get(config, "timestamps",
+                                                 "initial", default_initial))
+    result['offset_file'] = config_path(config_get(config, "logging",
+                                                   "offset_file", ".offset"),
+                                        result)
+
+    return result
+
 
+def config_get(config, section, param, default=None):
+    """ Get $param value from $section of $config with $default value.
 
-def config_path(rel_path):
+    If the parameter (or even section) is not presented in the config,
+      returns default value instead of throwing an exception (as
+      `config.get()` does).
+
+    :param config: config to read from
+    :param section: config section name
+    :param param: config parameter name
+    :param default: default value
+    :type config: ConfigParser object
+    :type section: string
+    :type param: string
+    :param default: object (any type)
+    :return: param value from config or default value
+    :rtype: object
+    """
+    if not isinstance(config, ConfigParser.ConfigParser):
+        raise TypeError("config_get() expects first parameter to be"
+                        " an instance of 'ConfigParser' (get '%s')."
+                        % config.__class__.__name__)
+    if type(section) != str and \
+            (sys.version_info.major > 2 or type(section) != unicode):
+        raise TypeError("config_get() expects second parameter to be"
+                        " string (get '%s')."
+                        % section.__class__.__name__)
+    if type(param) != str and \
+            (sys.version_info.major > 2 or type(param) != unicode):
+        raise TypeError("config_get() expects third parameter to be"
+                        " string (get '%s')."
+                        % section.__class__.__name__)
+
+    try:
+        result = config.get(section, param)
+    except ConfigParser.Error:
+        if default is not None:
+            sys.stderr.write("(INFO) Config: parameter '%s' (section '%s')"
+                             " is not configured. Using default value: %s\n"
+                             % (param, section, default))
+        result = default
```

Comment:
In PR#98 you said that for multiline log messages every line should be prepended with (ERROR|WARN|WARNING|INFO|DEBUG|INFO) (or only the first line, and all the others with (==)). I see (INFO) only on the first line. Should it be fixed? :)

Reply:
There\`s no `\n` in the end of first line, so it\`s just a matter of wrapping the code in 80 symbols ;)
See:
```
>>> sys.stderr.write("(INFO) Config: parameter '%s' (section '%s')"
...                  " is not configured. Using default value: %s\n"
...                  % ('str1', 'str2', 'str3'))
(INFO) Config: parameter 'str1' (section 'str2') is not configured. Using default value: str3
```

Reply:
Oh... Now I see. My apologies. :)